v3/src/hooks/use-drop-hint.ts

```diff
@@ -0,0 +1,76 @@
+import {useDndContext} from "@dnd-kit/core";
+import {useDataSetContext} from "./use-data-set-context";
+import {AxisPlace, axisPlaceToAttrPlace} from"../components/graph/models/axis-model";
+import {useDataConfigurationContext} from "../components/graph/hooks/use-data-configuration-context";
+import {getStringWithSwaps} from "../utilities/tokened-string";
+
+export interface IUseDropHintProps {
+  place?: AxisPlace,
+  dropId?: string
+}
+
+function determineBaseString(placeString: string,  dropType: string, existingType: string){
+  const stringMap = {
+    bottom: {
+      numeric: {
+        empty: "addToEmptyPlace",
+        existing: "replaceAttribute"
+      },
+      categorical: {
+        empty: "layoutPlotsSideBySide",
+        existing: "layoutPlotsSideBySide"
+      }
+    },
+    left: {
+      numeric: {
+        empty: "addToEmptyPlace",
+        existing: "replaceAttribute"
+      },
+      categorical: {
+        empty: "layoutPlotsVertically",
+        existing: "layoutPlotsVertically"
+      }
+    },
+    // in V2 a drop on empty plot puts attr on X axis ("addToEmptyX")
```

Comment:
Noticed slightly different behavior in v3 from v2 at the moment - not sure this is meant to be, or if the exact same behavior is expected to be in place down the road.  As noted:

v2 - first drop on empty plot puts the attr on the x axis
v3 - first drop on empty plot puts the attr in the legend

Reply:
Yes, down the road, in fact soon, if there are no attributes on the x or y axes, the drop in the plot area will go to the x-axis. (Nice observation!)

Reply:
makes sense, great